python/navigation/navigation_widget.py

```diff
@@ -0,0 +1,120 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Navigation widget containing previous, home and next buttons: [<][H][>].  The
+widget keeps track of the current location within a list and emits signals
+whenever the user navigates via one of the buttons.
+"""
+
+import sgtk
+from sgtk.platform.qt import QtCore, QtGui
+from .ui.navigation_widget import Ui_NavigationWidget
+
+class NavigationWidget(QtGui.QWidget):
+    """
+    Implementation of the NavigationWidget class
+    """
+
+    # Signal emitted whenever the user navigates using 
+    # the 'previous'/[<] or 'next'/[>] buttons.
+    navigate = QtCore.Signal(object)# destination to navigate to
+
+    # Signal emitted whenever the 'home'/[H] buttons is
+    # pressed
+    home_clicked = QtCore.Signal()
+
+    class _DestinationInfo(object):
```

Comment:
We should see if we still require support for Python 2.5. This could be replaced by a simple [namedtuple]( https://docs.python.org/2/library/collections.html#collections.namedtuple).